src/rpc/misc.cpp

```diff
@@ -243,6 +243,7 @@ static RPCHelpMan deriveaddresses()
             {
                 {"descriptor", RPCArg::Type::STR, RPCArg::Optional::NO, "The descriptor."},
                 {"range", RPCArg::Type::RANGE, RPCArg::Optional::OMITTED_NAMED_ARG, "If a ranged descriptor is used, this specifies the end or the range (in [begin,end] notation) to derive."},
+                {"require_checksum", RPCArg::Type::BOOL, RPCArg::Default{true}, "Fail if the descriptor checksum is invalid or missing."},
```

Comment:
Let's always fail if it's invalid. So if the user doesn't provide one and `require_checksum` is `false` we're happy, but if they provide a checksum, is must be valid.

Reply:
That is already the case.

```
$ ./bitcoin-cli -regtest getdescriptorinfo "combo(cTsWy7DsaN2uKgVueRHcUKAP3mCNzX5r77GsiLWw9C9zapjiK8tQ)"
{
  "descriptor": "combo(02fdbfbbfc35d0748ec58f628988090fce16985087e89639bccc4bfec4056323dc)#lc57gg0x",
  "checksum": "y9v74p5f",
  "isrange": false,
  "issolvable": true,
  "hasprivatekeys": true
}
$ ./bitcoin-cli -regtest deriveaddresses "combo(02fdbfbbfc35d0748ec58f628988090fce16985087e89639bccc4bfec4056323dc)#lc57gg0x"
[
  "mtrCy8QYfDU7tjPnkpWwWdQomuRfjptFy2",
  "mtrCy8QYfDU7tjPnkpWwWdQomuRfjptFy2",
  "bcrt1qjg7un9j86eld6y3sg0u0scmxedv6hued2z9m89",
  "2NDtkyew2s7qD1HYDXkNZTi1HY7nT7ZhB72"
]
$ ./bitcoin-cli -regtest deriveaddresses "combo(02fdbfbbfc35d0748ec58f628988090fce16985087e89639bccc4bfec4056323dc)#lc57gg0x" null false
[
  "mtrCy8QYfDU7tjPnkpWwWdQomuRfjptFy2",
  "mtrCy8QYfDU7tjPnkpWwWdQomuRfjptFy2",
  "bcrt1qjg7un9j86eld6y3sg0u0scmxedv6hued2z9m89",
  "2NDtkyew2s7qD1HYDXkNZTi1HY7nT7ZhB72"
]
$ ./bitcoin-cli -regtest deriveaddresses "combo(02fdbfbbfc35d0748ec58f628988090fce16985087e89639bccc4bfec4056323dc)#lc57gg1x" null false
error code: -5
error message:
Provided checksum 'lc57gg1x' does not match computed checksum 'lc57gg0x'
```

I've updated the description to reflect this.